accounts/service.go

```diff
@@ -531,14 +564,32 @@ func (s *InterceptorService) TrackPayment(id AccountID, hash lntypes.Hash,
 		return nil
 	}
 
-	// Okay, we haven't tracked this payment before. So let's now associate
-	// the account with it.
 	account.Payments[hash] = &PaymentEntry{
 		Status:     lnrpc.Payment_UNKNOWN,
 		FullAmount: fullAmt,
 	}
+
 	if err := s.store.UpdateAccount(account); err != nil {
-		return fmt.Errorf("error updating account: %w", err)
+		if ok {
+			return fmt.Errorf("error updating account: %w", err)
+		} else {
```

Comment:
Since the `if` is a `return`, we don't need to have the `else` and the indentation.

Reply:
Ah thanks!I noticed when fixing this that I had forgotten to return on the `else` case here, so I changed the order of the `if` check to instead check for `!ok` and then return the `err` below the check for both cases.